src/axom/core/Array.hpp

```diff
@@ -782,27 +908,20 @@ Array<T>::Array(T* data, IndexType num_elements, IndexType capacity)
 }
 
 //------------------------------------------------------------------------------
-template <typename T>
-Array<T>::Array(const Array& other, int allocator_id)
-  : m_data(nullptr)
-  , m_num_elements(0)
-  , m_capacity(0)
-  , m_resize_ratio(DEFAULT_RESIZE_RATIO)
-  , m_is_external(false)
-  , m_allocator_id(allocator_id)
+template <typename T, IndexType dim>
+Array<T, dim>::Array(const Array& other, int allocator_id)
+  : m_allocator_id(allocator_id)
 {
   initialize(other.size(), other.capacity());
   axom::copy(m_data, other.data(), m_num_elements * sizeof(T));
+  std::copy(other.m_dims, other.m_dims + dim, m_dims);
+  std::copy(other.m_strides, other.m_strides + dim, m_strides);
 }
```

Comment:
Should we favor `axom::copy` over `std::copy`, since it will account for the allocator?